Buddies/Controls/FilterSearchBar.swift

```diff
@@ -65,8 +67,24 @@ class FilterSearchBar : UISearchBar, UISearchBarDelegate {
         ])
     }
     
-    private func makeFilterButton(doing action: Selector, rounding corners: CACornerMask) -> UIButton {
-        let bttn = makeButton(saying: "Filter", doing: action)
+    func provideLoggedInUser(_ user: LoggedInUser?) {
+        self.user = user
+        
+        var filterSettings = user?.filterSettings ?? FilterSearchBar.defaultSettings
+        
+        // Convert to a filter state tuple
+        self.lastFilterState = (
```

Comment:
Why do we need to convert the stuff?

Reply:
1. We don't want to persist the text in the search bar
2. We can't store a tuple in Firestore so it has to change to an array